src/Components/Aspire.RabbitMQ.Client/RabbitMQEventSourceLogForwarder.cs

```diff
@@ -125,7 +125,20 @@ IEnumerator IEnumerable.GetEnumerator()
             return GetEnumerator();
         }
 
-        public int Count => 5;
+        public int Count
+        {
+            get
+            {
+                Debug.Assert(EventData.PayloadNames?.Count == 2 && EventData.Payload?.Count == 2);
+                Debug.Assert(EventData.PayloadNames[0] == "message");
+                Debug.Assert(EventData.PayloadNames[1] == "ex");
+
+                var exData = EventData.Payload[1] as IDictionary<string, object?>;
+                Debug.Assert(exData is not null && exData.Count == 4);
+
+                return string.IsNullOrEmpty(exData["InnerException"]?.ToString()) ? 4 : 5;
```

Comment:
Since this property gets accessed quite a bit (every time the indexer is accessed, and each loop through the `GetEnumerator()` method), we should move this logic into the constructor and store the count in a field. That way it doesn't need to get recomputed each time.

Reply:
Did the refactor! Thanks for the suggestion!